src/main/java/io/vlingo/wire/fdx/outbound/ManagedOutboundChannel.java

```diff
@@ -7,9 +7,14 @@
 
 package io.vlingo.wire.fdx.outbound;
 
+import reactor.core.publisher.Mono;
+
 import java.nio.ByteBuffer;
 
 public interface ManagedOutboundChannel {
   void close();
   void write(final ByteBuffer buffer);
+  default Mono<Void> writeAsync(final ByteBuffer buffer) {
```

Comment:
I am not sure we should use `reactor.core.publisher.Mono`, comming from a transient dependency, as a return parameter. I think it's better to use `io.vlingo.common.Completes`.

Reply:
I think so too. Any direction on how to go about it? I couldn't find a way do `Mono` -> `Completes`.

Reply:
Hi @buritos . Here is a simplified example of how to transform a `Mono` into `Completes.` Althought I am not sure that it is the most correct aproach. But it might help you to get the idea: 
```java
  public Completes<Void> write(final ByteBuffer buffer) {
    //Maybe we should get the scheduler from the actor that initializes the channel
    final Scheduler scheduler = new Scheduler();

    final Optional<RSocket> socket = prepareSocket();
    if (socket.isPresent()) {
      final RSocket rSocket = socket.get();
      //check if channel still open
      if (!rSocket.isDisposed()) {
        //Copy original buffer data because payload might not be sent immediately.
        ByteBuffer data = ByteBuffer.allocate(buffer.capacity());
        data.put(buffer);
        data.flip();

        //Initialize an empy Completes, that willl be completed by Mono
        final Completes<Void> result = Completes.using(scheduler);

        final Payload payload = DefaultPayload.create(data);
        rSocket.fireAndForget(payload).onErrorResume(throwable -> {
                  if (throwable instanceof ClosedChannelException) {
                    //close outbound channel
                    rSocket.dispose();
                    logger.error("Connection with {} closed", address, throwable);
                    return Mono.error(throwable);
                  } else {
                    logger.error("Failed write to {}, because: {}", address, throwable.getMessage(), throwable);
                    return Mono.empty();
                  }
                })
               .doFinally(signalType -> data.clear())
               .subscribe(
                       //Complete the result with either success or error
                       aVoid -> { result.with(aVoid); },
                       throwable -> { result.failed(); }
               );
        return result;
      } else {
        logger.warn("RSocket outbound channel for {} is closed. Message dropped", this.address);
        return Completes.withFailure();
      }
    } else {
      logger.debug("RSocket outbound channel for {} not ready. Message dropped", this.address);
      return Completes.repeatableUsing(scheduler);
    }
  }

```

Reply:
Hi @alexguzun, I have made some changes. I didn't find a good way to deal with `Collection<Completes> ` in `Outbound::broadcast`. I resorted to a latch which works pretty well but is not as fluent as the previous version with `Mono:: zipDelayError`. Also, I am not sure about the `Scheduler`, as you mentioned... it shouldn't be something the channel should create on its own. My view is that it should run on the same one as the `Mono`, but couldn't figure out a way to do this. Let me know what you think.